### PR TITLE
New darktable 2.2.X Lua script database_statisctics and steghide

### PR DESCRIPTION
**Pull request by klemmh**
Hallo,
here are to new darktable 2.2.X lua scripts.

Holger

### CODE REVIEW

File: contrib/database_statistics.lua [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+
+local nocolorlabels = dt.new_widget("label")
+{
+      label = _('Images without labels:\t\t') .. count_without,
+      ellipsize = "start",
+      halign = "start",
+}
+
+
+local function checkIfBinExists(bin)
```
Comment (supertobi) [inline contrib/database_statistics.lua]: We have a new utility class for this function, that you could use. Have a look into the geoToolbox.lua script:

local df = require "lib/dtutils.file"
df.check_if_bin_exists("PROGRAMM")

File: contrib/database_statistics.lua [hunk 2]
```diff
@@ -0,0 +1,441 @@
...
+  if (result) then
+    ret = true
+  else
+    dt.print_error(bin.." not found")
+    ret = false
+  end
+  return ret
+end
+
+local function checkIfDirExists(dir)
```
Comment (supertobi) [inline contrib/database_statistics.lua]: Could be replaced with:

dtutils_file.check_if_file_exists(filepath)

File: contrib/steghideexport.lua [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
...fish nofb","twofish ofb","arcfour stream","cast-256 cbc","cast-256 cfb","cast-256 ctr","cast-256 ecb","cast-256 ncfb","cast-256 nofb","cast-256 ofb","loki97 cbc","loki97 cfb","loki97 ctr","loki97 ecb","loki97 ncfb","loki97 nofb","loki97 ofb","rijndael-192 cbc","rijndael-192 cfb","rijndael-192 ctr","rijndael-192 ecb","rijndael-192 ncfb","rijndael-192 nofb","rijndael-192 ofb","saferplus cbc","saferplus cfb","saferplus ctr","saferplus ecb","saferplus ncfb","saferplus nofb","saferplus ofb","wake stream","des cbc","des cfb","des ctr","des ecb","des ncfb","des nofb","des ofb","rijndael-256 cbc","rijndael-256 cfb","rijndael-256 ctr","rijndael-256 ecb","rijndael-256 ncfb","rijndael-256 nofb","rijndael-256 ofb","serpent cbc","serpent cfb","serpent ctr","serpent ecb","serpent ncfb","serpent nofb","serpent ofb","xtea cbc","xtea cfb","xtea ctr","xtea ecb","xtea ncfb","xtea nofb","xtea ofb","blowfish cbc","blowfish cfb","blowfish ctr","blowfish ecb","blowfish ncfb","blowfish nofb","blowfish ofb","enigma stream","rc2 cbc","rc2 cfb","rc2 ctr","rc2 ecb","rc2 ncfb","rc2 nofb","rc2 ofb","tripledes cbc","tripledes cfb","tripledes ctr","tripledes ecb","tripledes ncfb","tripledes nofb","tripledes ofb",
```
Comment (supertobi) [inline contrib/steghideexport.lua]: Perhaps add a comment where this list comes from.

File: contrib/steghideexport.lua [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+    entrypassword,
+    label_enc,
+    enc_combobox,
+    label_path,
+    file_chooser_button,    
+}
+
+
+
+local function checkIfBinExists(bin)
```
Comment (supertobi) [inline contrib/steghideexport.lua]: same as above

Comment (supertobi) [inline contrib/steghideexport.lua]: @klemmh 
I haven't heard anything from you. What is the status of this scripts?

**Comment by txelu** [comment]
I'm new to this but I had to modify `include_all.lua` to enable API 4.0.0 and make it work with my DT 2.2.x. I changed from the code to check the version from:
`dt.configuration.check_version(...,{3,0,0})`
to:
`dt.configuration.check_version(...,{3,0,0},{4,0,0})`
I think that change is missing to complete the migration to API 4.0.0.

**Comment by klemmh** [comment]
My script versions:
script version 0.X -> Designed for Darktable 2.0.X
script version 2.X -> Designed for Darktable 2.2.X

Please use the version for DT 2.0.X at the end of the homepage

http://www.multimedia4linux.de/index.php/bildbearbeitung/darktable/darktable-erweitern-mit-lua-skripten



Am Freitag, 17. März 2017, 10:31:30 CET schrieb txelu:
> I'm new to this but I had to modify `include_all.lua` to enable API 4.0.0
> and make it work with my DT 2.2.x. I changed from the code to check the
> version from: `dt.configuration.check_version(...,{3,0,0})`
> to:
> `dt.configuration.check_version(...,{3,0,0},{4,0,0})`
> I think that change is missing to complete the migration to API 4.0.0.

**Comment by txelu** [comment]
Thanks a lot for your answer. It has been a nice discovery your site. I will use it to practice my German too :)

**Comment by supertobi** [comment]
@klemmh 
As you have not reacted on my review it looks like that you are not interested in maintaining your script here I'll close this pull request. Feel free to create a new one or reopen it if this assumption was wrong.

**Pull request closed without merging**